src/d3d9/d3d9_device.cpp

```diff
@@ -6696,34 +6700,99 @@ namespace dxvk {
 
 
   void D3D9DeviceEx::BindFramebuffer() {
-    constexpr size_t DsvIndex = caps::MaxSimultaneousRenderTargets;
-
     m_flags.clr(D3D9DeviceFlag::DirtyFramebuffer);
 
     DxvkRenderTargets attachments;
 
     bool srgb = m_state.renderStates[D3DRS_SRGBWRITEENABLE];
 
-    // Some games break if render targets that get disabled using the color write mask
-    // end up shrinking the render area. So we don't bind those.
-    // (This impacted Dead Space 1.)
+    // The extents and sample counts of all render targets need to match.
+    // There's a few exceptions for mismatching extents of depth stencil surfaces:
+    //   - It is allowed to be larger than RT0.
+    //   - It is allowed to be smaller than RT0 IF only one render target is bound, RT0 has the NULL format
+    //     or the extents 1x1 and the color write mask for RT0 is 0.
```

Comment:
I wish we actually had a test suite for these weird cases... I regret not doing that years ago. :(

Reply:
Yeah I should've tested the exact behavior years ago.
I *should* also write and contribute some Wine tests.